### PR TITLE
Adjust key regex and centralize patterns

### LINKED ISSUE #1876: Keystring regular expression too restrictive
I've been having some trouble with the keystring regular expression rejecting values.  For example, it will reject values like `double__underscore` and `tier_2`.  I suggest maybe `/^[a-z]+[_a-z0-9]*$/` instead.  I think it might be good to centralize our patterns in a shared file, too.

### PR DESCRIPTION
**Pull request by petertgiles**
This branch adjusts the keystring regular expression pattern to be less restrictive and centralizes the key and phone expressions into a shared constants file.

Closes #1876

### CODE REVIEW

File: frontend/common/src/constants/regularExpressions.tsx [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export const keyString = /^[a-z]+[_a-z0-9]*$/;
+export const phoneNumber = /^\+[1-9]\d{1,14}$/;
```
Comment (patcon) [inline frontend/common/src/constants/regularExpressions.tsx:2]: Clearer and less typing to just keep the variable consistent and avoid aliasing?

```suggestion
export const keyStringRegex = /^[a-z]+[_a-z0-9]*$/;
export const phoneNumberRegex = /^\+[1-9]\d{1,14}$/;
```

Comment (patcon) [reply frontend/common/src/constants/regularExpressions.tsx:2]: Further: I've gotten in the habit of adding a link to a good testing tool, with some good/bad strings, as a way to explore the regex more intuitively :)

```suggestion
# Links to regex experimentation tool are provided for each regex pattern.
# Please feel free to "Update" and liberally add candidate test strings.
# If you make changes, they'll be most discoverable by updating the link below.

# See: https://regex101.com/r/TEDtDo/1
export const keyStringRegex = /^[a-z]+[_a-z0-9]*$/;
# See: https://regex101.com/r/EAZ8ha/1
export const phoneNumberRegex = /^\+[1-9]\d{1,14}$/;
```

Comment (petertgiles) [reply frontend/common/src/constants/regularExpressions.tsx:2]: These are great suggestions!  I'm putting them in.

**Review by yonikid15 — APPROVED** [review]
Nice work again Peter. Also, love the link to the regex examples, thanks Pat.

File: frontend/common/src/constants/regularExpressions.tsx (hunk 1 above)
Comment (tristan-orourke) [reply frontend/common/src/constants/regularExpressions.tsx]: Nice, this is a great idea!

**Pull request merged**